source/geod24/concurrency.d

```diff
@@ -985,6 +986,161 @@ protected:
         bool notified;
     }
 
+    class Channel (T)
```

Comment:
Is the intention of this `Channel` implementation be used with multiple threads? It seems that way from the usage of Mutex. However, there are no multithreaded tests which use the channel in this PR. Only multi-fiber tests which use the same thread. So the locks aren't really being tested properly here AFAICT.

Also, if a Channel will be shared across multiple threads then it should likely be marked as `shared`. I'm cc'ing @Geod24 for input here.

Reply:
Yes, my intention was Channels to be thread safe but only to be used between Fibers. The tests are lacking, will add more.